guava/src/com/google/common/base/MoreObjects.java

```diff
@@ -57,6 +56,43 @@
   }
 
   /**
+   * If the input is null, it returns a default value.
+   * Otherwise it applies a function to it.
+   * <p/>
+   * Specification by example:
+   * <pre>{@code
+   *   // Returns "null"
+   *   String name = null;
+   *   MoreObjects.ifNonNull(name, new Function<String, String>() {
+   *     public String apply(String n) { return n.toUpperCase(); }
+   *   });
+   * ;
+   * }</pre>
+   * or in Java 8 syntax, with non-null input, and default value
+   * <pre>{@code
+   *   // Returns "NAME"
+   *   String name = "name";
+   *   MoreObjects.ifNonNull(name, n -> n.toUpperCase(), "");
+   * }</pre>
+   *
+   * @param input        The value that should be null-checked
+   * @param thenFunction The function to apply to the non-null input
+   * @param elseValue    The value that should be used in case the input was null
+   * @return if input is null, return the default value, otherwise apply a function to it
+   * @since 19.0
+   */
+  @Nullable
+  public static <T, R> R ifNonNull(@Nullable T input, @Nonnull Function<T, R> thenFunction, @Nullable R elseValue) {
```

Comment:
Another overload with a `Supplier` for the `elseValue` could be useful, in order to make lazy evaluation possible.